### DISCUSSION TITLE
Can I use find method with relation fields??

### DISCUSSION THREAD

**Discussion by enzzoperez** [strapi/strapi#7306] [Legacy Topics]
Hi!, I have a model similar to this, where `instructor` is a `video` relation field 

```
{
    id: '234b2hj3jk23j',
    name: 'video 1',
    instructor: {
        name: 'instructor one',
        pic: 'pic.com/pic.jpg'
    }
}
```

In a custom controller, can I use `find` method to filter by instructor name?, something like `strapi.query('videos').find({ instructor: {name: 'instructor one'} })` but this doesnt work

Thanks!!

**Comment by derrickmehaffy** [comment]
Hello!

Due to how the deep filtering works you will need to combine the sub-field and main-field together, something like this:

`strapi.query('video').find({ 'instructor.name': 'instructor one'});`

Please let me know if that works, and if so, please feel free to mark this as the answer :smiley:

**Reply by derrickmehaffy** [reply]
Something to note, when you are using the `strapi.query()` it will need to be the internal name of the model, you can find this in your `./api/*/models/*.settings.json` as the `info.name` key not the `collectionName` which is what is used in the database and for the routes. (Typically this is just the singular version of the model name)

**Reply by enzzoperez** [reply]
Thank you for response, I will try with that

**Reply by enzzoperez** [reply]
@derrickmehaffy  Yay!, that works!, but I get a warning `Deep filtering queries should be used carefully (e.g Can cause performance issues).
When possible build custom routes which will in most case be more optimised.` , Is that the best way? or maybe I would have to get the results with `strapi.query('video').find({});` and then iterate with `.filter(v=>v.instructor === 'instructor one')` ?

**Reply by derrickmehaffy** [reply]
That warning just comes from deep filtering in general (and there is no way to disable it) If you wanted to get rid of that warning you would want to build a custom query that directly interacts with bookshelf/mongoose: https://strapi.io/documentation/v3.x/concepts/queries.html#custom-queries
